pytorch_lightning/callbacks/model_summary.py

```diff
@@ -62,11 +62,10 @@ def on_pretrain_routine_start(self, trainer: "pl.Trainer", pl_module: "pl.Lightn
 
     @staticmethod
     def summarize(
-        summary_data: List[List[Union[str, List[str]]]],
+        summary_data: List[Tuple[str, List[str]]],
```

Comment:
@kaushikb11 - if this set of arguments is going to grow, should we use a dataclass instead? that way the `summarize` signature doesn't change, and we can still add more arguments over time w/o breaking backwards compatibility or making everything optional.

Reply:
@ananthsub The idea makes sense but I am not sure what additional arguments we will be planning to add in the future.